src/Angor/Avalonia/Angor.Wallet/Infrastructure/Impl/AesWalletEncryption.cs

```diff
@@ -0,0 +1,90 @@
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Json;
+using Angor.Wallet.Infrastructure.Interfaces;
+using CSharpFunctionalExtensions;
+
+namespace Angor.Wallet.Infrastructure.Impl;
+
+public class AesWalletEncryption : IWalletEncryption
+{
+    private const int Iterations = 100000;
+    private const int KeySize = 256;
+
+    public async Task<Result<WalletData>> Decrypt(EncryptedWallet encryptedWallet, string encryptionKey)
```

Comment:
I prefer to have the word async in the end of an async method, perhaps we should bring this up with the others and have a consensus.

Reply:
I see your point about the Async suffix. Yes, it's in Microsoft's guidelines, but I'm not a fan of it for practical reasons - it creates redundancy and unnecessarily long names, especially in interfaces where almost every method returns a Task.
That said, if we want to standardize this across our codebase, I'm fine changing it. Should we bring this up with the team to decide on a consistent approach?

Reply:
Several popular .NET projects (like MediatR, Octokit, NServiceBus) don't follow this convention either. But I'm happy to go with whatever standard we decide on as a team.

Reply:
Keep it like this but let's bring it up in the next team meeting so we are all on the same page - consistency is more important